From b7fd50f4c1db7ac887ff13af728314bb2eb7a620 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Tue, 23 Apr 2013 00:33:15 +0200
Subject: [PATCH 55/65] r92su: only allow legacy monitor channels for now

It might be possible to set HT40+/- and HT20
with mp_mode, but the details are unknown.

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/main.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/wireless/r92su/main.c
+++ b/drivers/net/wireless/r92su/main.c
@@ -817,6 +817,9 @@ static int r92su_set_monitor_channel(str
 	struct r92su *r92su = wiphy_priv(wiphy);
 	int err = -EAGAIN;
 
+	if (cfg80211_get_chandef_type(chandef) != NL80211_CHAN_NO_HT)
+		return -EOPNOTSUPP;
+
 	mutex_lock(&r92su->lock);
 
 	r92su->current_channel = chandef->chan;
